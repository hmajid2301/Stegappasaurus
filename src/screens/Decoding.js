import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';

import { toggleTheme, changeTheme } from '../actions';
import COLORS from '../themes';


class Decoding extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
    changeTheme: PropTypes.func.isRequired,
    toggleTheme: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.navigation.addListener('willFocus', (payload) => {
      const routeName = payload.state.key;
      this.props.changeTheme(routeName);
      this.props.toggleTheme(COLORS.primary);
    });
  }

  render() {
    return (
      <View>
        <Text> Decoding </Text>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  changeTheme: route => dispatch(changeTheme(route)),
  toggleTheme: color => dispatch(toggleTheme(color)),
});

export default connect(null, mapDispatchToProps)(Decoding);
