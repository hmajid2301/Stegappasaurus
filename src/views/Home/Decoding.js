import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';

import { togglePrimaryColor } from '~/redux/actions';
import { PRIMARY_COLORS } from '~/util/constants';

import styles from './Decoding/styles';


class Decoding extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.object.isRequired,
    togglePrimaryColor: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.navigation.addListener('willFocus', () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.BLUE.name);
    });
  }

  render() {
    const { theme } = this.props.screenProps;
    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <Text> Decoding </Text>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  togglePrimaryColor: color => dispatch(togglePrimaryColor(color)),
});

export default connect(null, mapDispatchToProps)(Decoding);
