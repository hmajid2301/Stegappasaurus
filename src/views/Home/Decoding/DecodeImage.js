import PropTypes from 'prop-types';
import React, { Component } from 'react';

import ImageMessage from '~/views/Home/ImageMessage';
import ImageProgressCircle from '~/views/Home/ImageProgressCircle';
import { colors } from '~/util/styles';


export default class DecodeImage extends Component {
  constructor(props) {
    const { navigation } = props;
    const uri = navigation.getParam('uri', 'NO-ID');

    super(props);
    this.state = {
      isDecoded: false,
      photo: uri,
    };
  }

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.object.isRequired,
  }

  decoded = () => {
    this.setState({ isDecoded: true });
  }

  render() {
    const { theme } = this.props.screenProps;

    if (this.state.isDecoded) {
      return (
      <ImageMessage message={'Temp'} photo={this.state.photo}/>
      );
    }

    return (
      <ImageProgressCircle
        action={this.decoded}
        photo={this.state.photo}
        primaryColor={colors.secondary}
        theme={theme}
      />
    );
  }
}
