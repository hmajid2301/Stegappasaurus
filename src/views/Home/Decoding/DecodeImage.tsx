import React, { Component } from "react";
import { NavigationScreenProp } from "react-navigation";

import { colors } from "~/util/styles";
import ImageMessage from "~/views/Home/ImageMessage";
import ImageProgressCircle from "~/views/Home/ImageProgressCircle";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: {
      background: string;
      color: string;
      isDark: boolean;
    };
  };
}

interface IState {
  isDecoded: boolean;
  photo: string;
}

export default class DecodeImage extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");

    this.state = {
      isDecoded: false,
      photo: uri
    };
  }

  public render() {
    const { theme } = this.props.screenProps;

    if (this.state.isDecoded) {
      return <ImageMessage message={"Temp"} photo={this.state.photo} />;
    }

    return (
      <ImageProgressCircle
        action={this.decoded}
        photo={this.state.photo}
        primaryColor={colors.secondary}
        theme={theme}
      />
    );
  }

  private decoded = () => {
    this.setState({ isDecoded: true });
  };
}
