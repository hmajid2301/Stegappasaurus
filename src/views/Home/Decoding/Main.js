import { ImagePicker, Permissions } from 'expo';
import { Icon } from 'native-base';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';

import { togglePrimaryColor } from '~/redux/actions';
import { PRIMARY_COLORS } from '~/util/constants';

import styles from './Main/styles';


class Decoding extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.shape({
      theme: PropTypes.shape({
        isDark: PropTypes.bool.isRequired,
        background: PropTypes.string.isRequired,
        color: PropTypes.string.isRequired,
      }),
    }),
    togglePrimaryColor: PropTypes.func.isRequired,
  };

  componentDidMount = () => {
    this.props.navigation.addListener('willFocus', () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.BLUE.name);
    });
  }

  getPhotoFromCameraRoll = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const result = await ImagePicker.launchImageLibraryAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      throw new Error('Camera Roll permission not granted');
    }
  }

  selectPhotoToEncode = (uri) => {
    this.props.navigation.navigate('DecodeImage', { uri });
  }

  render() {
    const { theme } = this.props.screenProps;
    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <TouchableOpacity onPress={this.getPhotoFromCameraRoll} style={styles.button}>
          <Icon name='photo' style={styles.icon} type='FontAwesome' />
        </TouchableOpacity>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  togglePrimaryColor: color => dispatch(togglePrimaryColor(color)),
});

export default connect(null, mapDispatchToProps)(Decoding);
