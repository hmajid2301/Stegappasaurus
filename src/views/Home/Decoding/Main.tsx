import * as ImagePicker from "expo-image-picker";
import * as Permissions from "expo-permissions";
import * as React from "react";
import { TouchableOpacity, View } from "react-native";
import { Icon } from "react-native-elements";
import { NavigationScreenProp } from "react-navigation";
import { connect } from "react-redux";
import { Dispatch } from "redux";

import { ITheme, PrimaryColorNames } from "@types";
import PhotoAlbumList from "~/components/PhotoAlbumList";
import Snackbar from "~/components/Snackbar";
import { PRIMARY_COLORS } from "~/constants";
import { togglePrimaryColor } from "~/redux/actions";

import styles from "./Main/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
  togglePrimaryColor: (primaryColor: PrimaryColorNames) => void;
}

class Main extends React.Component<IProps, {}> {
  public render() {
    const { theme } = this.props.screenProps;
    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <View style={styles.buttonsRow}>
          <TouchableOpacity
            onPress={this.getPhotoFromCameraRoll}
            style={styles.button}
          >
            <Icon name="photo" iconStyle={styles.icon} type="font-awesome" />
          </TouchableOpacity>
        </View>

        <View style={styles.photoListContainer}>
          <PhotoAlbumList onPhotoPress={this.selectPhotoToEncode} />
        </View>
      </View>
    );
  }

  public componentDidMount = () => {
    this.props.navigation.addListener("willFocus", () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.BLUE.name);
    });
  };

  private getPhotoFromCameraRoll = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === "granted") {
      const result = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: ImagePicker.MediaTypeOptions.Images
      });
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      Snackbar.show({
        text: "This app does not have permission to access the camera roll."
      });
    }
  };

  private selectPhotoToEncode = (uri: string) => {
    this.props.navigation.navigate("Progress", { uri });
  };
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  togglePrimaryColor: (colorName: PrimaryColorNames) =>
    dispatch(togglePrimaryColor({ color: colorName }))
});

export default connect(
  null,
  mapDispatchToProps
)(Main);
