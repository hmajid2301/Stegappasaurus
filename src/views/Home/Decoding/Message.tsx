import * as React from "react";
import { NavigationScreenProp } from "react-navigation";

import ImageMessage from "~/components/ImageMessage";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
}

interface IState {
  isDecoded: boolean;
  message: string;
  photo: string;
}

class Progress extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");
    const message = navigation.getParam("message", "NO-ID");

    this.state = {
      isDecoded: false,
      message,
      photo: uri
    };
  }

  public render() {
    return (
      <ImageMessage
        editable={false}
        message={this.state.message}
        photo={this.state.photo}
      />
    );
  }
}

export default Progress;
