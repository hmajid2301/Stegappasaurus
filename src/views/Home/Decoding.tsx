import {
  createStackNavigator,
  NavigationRoute,
  NavigationScreenProp
} from "react-navigation";

import Main from "./Decoding/Main";
import Message from "./Decoding/Message";
import Progress from "./Decoding/Progress";

const DecodeNavigator = createStackNavigator({
  Main: {
    navigationOptions: {
      header: null
    },
    screen: Main
  },

  DecodingMessage: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Message
  },

  DecodingProgress: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Progress
  }
});

DecodeNavigator.navigationOptions = ({
  navigation
}: {
  navigation: NavigationScreenProp<NavigationRoute>;
}) => ({
  swipeEnabled: navigation.state.index === 0,
  tabBarVisible: navigation.state.index === 0
});

export default DecodeNavigator;
