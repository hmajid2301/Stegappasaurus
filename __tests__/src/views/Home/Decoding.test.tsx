import { shallow } from "enzyme";
import React from "react";

import Decoding from "~/views/Home/Decoding";

describe("Decoding: Match Snapshots", () => {
  test("1", () => {
    const component = shallow(<Decoding />);
    expect(component).toMatchSnapshot();
  });
});
