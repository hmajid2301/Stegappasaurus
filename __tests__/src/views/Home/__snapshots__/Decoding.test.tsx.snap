// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decoding: Match Snapshots 1 1`] = `
<Navigator
  onGestureBegin={[Function]}
  onGestureCanceled={[Function]}
  onGestureEnd={[Function]}
  onTransitionStart={[Function]}
/>
`;
